workers/social/lib/social/models/group/index.coffee

```diff
@@ -1466,32 +1467,62 @@ module.exports = class JGroup extends Module
           @addMember account, {}, callback  unless removeUserFromTeam
 
 
-  transferOwnership: permit
+  transferOwnership$: permit
     advanced: [
       { permission: 'grant permissions' }
       { permission: 'grant permissions', superadmin: yes }
     ]
-    success: (client, accountId, callback) ->
-      JAccount = require '../account'
+    success: (client, options, callback) ->
 
       { delegate } = client.connection
-      if delegate.getId().equals accountId
+      { accountId, currentPassword, slug } = options
+
+      clientAccountId = delegate.getId()
+
+      return callback new KodingError 'User is not selected!'  unless accountId
+
+      if clientAccountId.equals accountId
         return callback new KodingError 'You cannot transfer ownership to yourself, concentrate and try again!'
 
+      checkUserPassword delegate, currentPassword, (err) =>
+
+        return callback new KodingError err  if err
+
+        @transferOwnership { clientAccountId, accountId, slug }, callback
+
+
+  transferOwnership: (options, callback) ->
+
+    { clientAccountId, accountId, slug } = options
+
+    return callback new KodingError 'Account is not provided'  unless accountId
+
+    @fetchOwner (err, owner) =>
+
+      return callback err if err
+
+      if clientAccountId and JSON.stringify(clientAccountId) isnt JSON.stringify(owner.getId())
```

Comment:
and side note, if you need to compare those Id's you can use `owner.getId().equals clientAccountId` which is the correct way for comparison here.